### PR TITLE
91 build create event page

### PR DESCRIPTION
**Pull request by pratibhapaudel**
## Release Notes

- **New Features**
  - Introduced a new `AddEvent` component for creating events.
  - Added a rich text `TextEditor` component for event descriptions.
  - Implemented a custom hook `useCreateEventPage` to manage event creation form state.
  - Enhanced the `AddEventPage` interface for user-friendly event creation.
  - Added a new `InputLabel` component for improved form labeling.

- **Improvements**
  - Updated the `ButtonLink` in the `EventsList` to direct users to the new event creation page.
  - Enhanced global styles with Tailwind CSS for improved typography and animations.
  - Improved styling for the Quill editor to enhance user experience.

- **Dependencies**
  - Added new dependencies including `@next/env` and `react-quill` for text editing capabilities.

### CODE REVIEW

File: src/modules/events/EventsList.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ const EventsPage = () => {
         </div>
 
         <div className="mt-8 py-1 text-center">
-          <ButtonLink href="#">Add New Event</ButtonLink>
+          <ButtonLink href="events/add-event">Add New Event</ButtonLink>
```
[the remark this answers is not in the document]
Comment (Aaasisss) [reply src/modules/events/EventsList.tsx]: please check this @pratibhapaudel

File: src/app/events/add-event/page.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { AddEventPage } from "@/modules/add-event";
+
```
Comment (Aaasisss) [inline src/app/events/add-event/page.tsx:2]: please add some metadata, similar to this
```ts
export const metadata: Metadata = {
  title: "Xplorers - Learn share and grow",
  description: "Xplorers is a group of talented tech nerds",
};
```

File: src/components/ui/label.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+interface InputLabelProps {
+  name: string;
+  required: boolean;
+  html_for: string;
```
Comment (Aaasisss) [inline src/components/ui/label.tsx:4]: consider using `camelCase` - `htmlFor` . it is widely used for Javascript ecosystem.

**Review by Aaasisss — APPROVED** [review]
looking good for now!

**Comment by pratibhapaudel** [comment]
Merged. Thank you

**Pull request merged**